tests/test_sparse_matrix_device.cu

```diff
@@ -245,3 +245,130 @@ BOOST_AUTO_TEST_CASE(distributed_mv)
   mfmg::ASSERT_CUSPARSE(cusparse_error_code);
   cusparse_handle = nullptr;
 }
+
+template <typename ScalarType>
+std::tuple<std::vector<ScalarType>, std::vector<int>, std::vector<int>>
+copy_sparse_matrix_to_host(
+    mfmg::SparseMatrixDevice<ScalarType> const &sparse_matrix_dev)
+{
+  std::vector<ScalarType> val =
+      copy_to_host(sparse_matrix_dev.val_dev, sparse_matrix_dev.local_nnz());
+
+  std::vector<int> column_index = copy_to_host(
+      sparse_matrix_dev.column_index_dev, sparse_matrix_dev.local_nnz());
+
+  std::vector<int> row_ptr =
+      copy_to_host(sparse_matrix_dev.row_ptr_dev, sparse_matrix_dev.m() + 1);
+
+  return std::make_tuple(val, column_index, row_ptr);
+}
+
+BOOST_AUTO_TEST_CASE(mmult)
+{
+  MPI_Comm comm = MPI_COMM_WORLD;
+  unsigned int const comm_size = dealii::Utilities::MPI::n_mpi_processes(comm);
+  if (comm_size == 1)
+  {
+    cusparseHandle_t cusparse_handle = nullptr;
+    cusparseStatus_t cusparse_error_code;
+    cusparse_error_code = cusparseCreate(&cusparse_handle);
+    mfmg::ASSERT_CUSPARSE(cusparse_error_code);
+
+    // Build the sparsity pattern
+    dealii::SparsityPattern sparsity_pattern;
+    unsigned int const size = 30;
+    std::vector<std::vector<unsigned int>> column_indices(size);
+    for (unsigned int i = 0; i < size; ++i)
+    {
+      std::vector<unsigned int> indices;
+      std::default_random_engine generator(i);
+      std::uniform_int_distribution<int> distribution(0, size - 1);
+      for (unsigned int j = 0; j < 5; ++j)
+        indices.push_back(distribution(generator));
+      indices.push_back(i);
+
+      std::sort(indices.begin(), indices.end());
+      indices.erase(std::unique(indices.begin(), indices.end()), indices.end());
```

Comment:
Instead of doing post-processing here, you could do something like this:
```c++
int cur = 0;
for (unsigned int j = 0; j < 5; ++j) {
  std::uniform_int_distribution<int> distribution(0, size - 1 - (5-j) - cur);
  indices.push_back(distribution(generator));
  cur = indices.back();
}
```
Essentially, you will guarantee that there are always 5 indices, and that they are sorted, but still in random places. I think that math properties may be slightly different, but for the test it should not matter. 

Or, you can think of this as an overkill ;)

Reply:
I don't think it works. For example:
```C++
size = 100;
j = 0;
indices[0] = 1;  // 1 is in (0, 99)
cur = 1;
j = 1;
indices[1] = 0; // 0 is in (0, 93);
```

Reply:
Sorry, you are right, I meant `distribution(cur+1, size - 1 - (5-j)-cur)`. This way, first index would be in the range `[0, size-1 - 5]`, second in `[first+1, size-1-4]`, etc.